### PR TITLE
Use single controller method to add transactions

### PR DESCRIPTION
**Pull request by matthewwalsh0**
## Explanation

Combine the `newUnapprovedTransaction` and `addUnapprovedTransaction` methods in the `TransactionController` into a single `addTransaction` method that returns the transaction metadata and a nested promise that resolves when the transaction is published.

This aligns with the core `TransactionController` API to support future migration.

This also includes the following low level refactors:

- Simplify arguments to `txParams` and an options bag.
- Change existing `addTransaction` method to private.
- Refactor related unit tests and restructure for simplicity using nested describes.
- Update `newUnapprovedTransaction` wrapper in `MetaMaskController`.
- Create `addTransactionAndWaitForPublish` wrapper in `MetaMaskController` to support frontend actions.

## Manual Testing Steps

Verify all existing transaction support including dApps and swaps.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: app/scripts/metamask-controller.js
Comment (vinistevam) [inline app/scripts/metamask-controller.js:3524]: :1st_place_medal:

File: app/scripts/metamask-controller.js
Comment (vinistevam) [inline app/scripts/metamask-controller.js:3525]: :1st_place_medal:

**Review by vinistevam — APPROVED** [review]

File: app/scripts/metamask-controller.js
Comment (matthewwalsh0) [reply app/scripts/metamask-controller.js:3528]: I'd argue it would only be simpler if you wanted to wait for the result.

This is the central benefit of this new method signature, as the caller can now choose whether they are only concerned with adding a transaction but don't care about approval or submission (the previous `addUnapprovedTransaction`), or they can wait for approval and submission (the previous `newUnapprovedTransaction`).

This is also the existing return type in the core `TransactionController` meaning the central method signature is now much more aligned going forward.

There's an argument that awaiting the result could be conditional inside the method, but that would diverge from the core `TransactionController`, plus greatly limit the clarity of the return value as what you're awaiting would be dynamic and so less predictable, plus you'd have an undefined value for the transaction hash if you chose not to await the result for example.

**Review by cryptotavares — APPROVED** [review]

**Pull request merged**
